yolapy/models/subscription.py

```diff
@@ -5,54 +5,42 @@ class Subscription(object):
 
     """Construct a Subscription.
 
-    The data in a Subscription instance does not persist until it has
-    been saved.
-
-    :param auto_renew: Bool
-    :param billing_date: Datetime
-    :param deprovision_date: Datetime
-    :param expiry_date: Datetime
-    :param start_date: Datetime
+    :param id: Str, subscription's 32 char ID
+    :param created_at: Str, datetime when subscription was created
+    :param updated_at: Str, datetime when subscription was updated last time
+    :param partner_id: Str, ID of the sub's partner
+    :param user_id: Str, 32 character ID of the sub's user
     :param status: Str, subscription status
     :param term: Str, subscription billing term
-    :param id: Str, subscription's 32 char ID
-    :param properties: Dict, subscription properties. If unset will be
-        defaulted to the client's configured username:
-        ``{"partner_id": "<CLIENT_USERNAME>"}``.
-    :param sku: Int
+    :param sku: Str, SKU of related product
     :param type: Str, subscription type
-    :param user_id: Str, 32 character ID of the sub's user
+    :param start_date: Str, subscription's start date
+    :param expiry_date: Str, subscription's expiry date
+    :param billing_date: Str, subscription's billing date
+    :param deprovision_date: Str, subscription's deprovision date
+    :param auto_renew: Bool
+    :param properties: Dict, subscription's properties
 
     :return: Subscription
     :rtype: yolapy.models.Subscription
 
     """
 
-    def __init__(
-            self, auto_renew=False, billing_date=None, deprovision_date=None,
-            expiry_date=None, id=None, properties=None, sku=None,
-            start_date=None, status=None, term=None, type=None, user_id=None):
-        self.client = Yola()
+    _fields = (
+        'id', 'created_at', 'updated_at', 'partner_id', 'user_id', 'status',
+        'term', 'sku', 'type', 'start_date', 'expiry_date', 'billing_date',
+        'deprovision_date', 'auto_renew', 'properties')
 
-        self.auto_renew = auto_renew
-        self.billing_date = billing_date
-        self.deprovision_date = deprovision_date
-        self.id = id
-        self.properties = properties or {}
-        self.properties.setdefault('partner_id', self.client.username)
-        self.sku = sku
-        self.type = type
-        self.user_id = user_id
-        self.expiry_date = expiry_date
-        self.start_date = start_date
-        self.status = status
-        self.term = term
+    def __init__(self, **fields):
+        self.client = Yola()
+        for field_name in self._fields:
+            setattr(self, field_name, fields[field_name])
```

Comment:
I share @ekoval's concern here. We shouldn't be prevented from using the model outside of the context of getting it from the service though. I very often use demands-based models in that way, especially in tests. I think we should use `.get(field_name)` here so the models behave the way most models we are used to using (e.g. you can initialize it with all, some, or no data, set some fields later if you want, then save).